### ISSUE TITLE
Can not use @JoinColumns

### ISSUE THREAD

**Issue by deverlex** [ebean-orm/ebean#3440]
My codes are below:
```java
@ManyToOne
@JoinColumns({
    @JoinColumn(name="org_id", referencedColumnName ="org_id", foreignKey = @ForeignKey(ConstraintMode.NO_CONSTRAINT)),
    @JoinColumn(name = "product_id", referencedColumnName = "id")
})
private Product product;
```
When I build and run project, it has been throw an exception
```console
ERROR  io.ebean.internal - Error in deployment
java.lang.NullPointerException: Cannot invoke "io.ebeaninternal.server.deploy.id.ImportedId.isScalar()" because "this.importedId" is null
	at io.ebeaninternal.server.deploy.BeanPropertyAssocOne.initialiseAssocOne(BeanPropertyAssocOne.java:128)
	at io.ebeaninternal.server.deploy.BeanPropertyAssocOne.initialise(BeanPropertyAssocOne.java:109)
	at io.ebeaninternal.server.deploy.BeanDescriptor.initialiseOther(BeanDescriptor.java:525)
	at io.ebeaninternal.server.deploy.BeanDescriptorManager.initialiseAll(BeanDescriptorManager.java:498)
	at io.ebeaninternal.server.deploy.BeanDescriptorManager.deploy(BeanDescriptorManager.java:301)
	at io.ebeaninternal.server.core.InternalConfiguration.<init>(InternalConfiguration.java:127)
	at io.ebeaninternal.server.core.DefaultContainer.createServer(DefaultContainer.java:118)
	at io.ebeaninternal.server.core.DefaultContainer.createServer(DefaultContainer.java:33)
	at io.ebean.DatabaseFactory.createInternal(DatabaseFactory.java:135)
```

I want use it for
```sql
select *
from sub_products sp
join products p on p.org_id = sp.org_id and p.id = sp.product_id
```

Ebean version: 13.25.2
Java version 1.8.0_381

**Comment by rob-bygrave** [comment]
I'd look to remove `foreignKey = @ForeignKey(ConstraintMode.NO_CONSTRAINT)`.  Ebean has a `@DbForeignKey` that you would look to use instead that would put on the `product` field.

Are you sure about the use of `@JoinColumns` and that the primary key for `Product` includes those 2 columns?  Can you show us what `@Product` is like and specifically what its `@Id` mapping is?

**Comment by deverlex** [comment]
> I'd look to remove `foreignKey = @ForeignKey(ConstraintMode.NO_CONSTRAINT)`. Ebean has a `@DbForeignKey` that you would look to use instead that would put on the `product` field.
> 
> Are you sure about the use of `@JoinColumns` and that the primary key for `Product` includes those 2 columns? Can you show us what `@Product` is like and specifically what its `@Id` mapping is?
Hi @rob-bygrave 

My `Product` entity has a `@Id` is `id`. I has been create an index on `orgId` and I want to use it for improve my query performance.
I defined `Product` in `SubProduct` and when I load `Product` from `SubProduct`, I want to it use 2 columns for join
```
@Id
private Long id;
private Long orgId;
```

**Comment by deverlex** [comment]
I found a image for this sample
![image](https://github.com/ebean-orm/ebean/assets/8696532/d1b24e95-7911-4116-81f9-cc105af3eb5f)

**Comment by rob-bygrave** [comment]
Well no 
(A) That does not work with Ebean
(B) It's a really bad example

What we are mapping with `@ManyToOne` is the database foreign key column(s). The database foreign key relates to the **_Primary Key_** of the related table.  It does not map to ... "the primary key column(s) plus some other arbitrary columns that I'd also like to map".

In this example the foreign key column is `product_id` and that references the `id` column of the `product` table.

> I has been create an index on orgId and I want to use it for improve my query performance.

In terms of the relational model we have the "Logical Model" (Primary keys, Foreign keys etc) and the "Physical Model" (Indexes, tablespaces etc).  One of the main strengths of the relational model is that there is this separation.

When we are mapping with `@Id` and `@ManyToOne` we are mapping to the relational Logical model and that means that Indexes are unrelated / independent to that.  We can add or remove indexes independently of the logical model / mapping (and we typically do so for performance reasons).

Most of the time we do want an index on foreign key columns but its not strictly required.


-------------
TLDR: The `@ManyToOne` join columns MUST map to the associated Primary key columns. You may have a query performance issue but you need to show us that query and probably the query plan and show us what that problem looks like.

**Comment by deverlex** [comment]
> Well no (A) That does not work with Ebean (B) It's a really bad example
> 
> What we are mapping with `@ManyToOne` is the database foreign key column(s). The database foreign key relates to the **_Primary Key_** of the related table. It does not map to ... "the primary key column(s) plus some other arbitrary columns that I'd also like to map".
> 
> In this example the foreign key column is `product_id` and that references the `id` column of the `product` table.
> 
> > I has been create an index on orgId and I want to use it for improve my query performance.
> 
> In terms of the relational model we have the "Logical Model" (Primary keys, Foreign keys etc) and the "Physical Model" (Indexes, tablespaces etc). One of the main strengths of the relational model is that there is this separation.
> 
> When we are mapping with `@Id` and `@ManyToOne` we are mapping to the relational Logical model and that means that Indexes are unrelated / independent to that. We can add or remove indexes independently of the logical model / mapping (and we typically do so for performance reasons).
> 
> Most of the time we do want an index on foreign key columns but its not strictly required.
> 
> TLDR: The `@ManyToOne` join columns MUST map to the associated Primary key columns. You may have a query performance issue but you need to show us that query and probably the query plan and show us what that problem looks like.

I will `@EmbeddedId` for `id` and `orgId`, I hope it will work successfully

**Comment by rob-bygrave** [comment]
Hmm ok, to me it seems like we are missing some information around the motivation for doing this ... like table partitioning or some example of a query that isn't performing and some explain plan or something like that.
